### DISCUSSION TITLE
auto increment does not work

### DISCUSSION THREAD

**Discussion by Sed9yDataBank** [typegoose/auto-increment#10] [Q&A]
I want to auto increment progressionIndex whenever new progression is created.
using typegoose + typegraphql
my code:

```typescript
@ObjectType()
export class Progression {
  @Field()
  @prop()
  progressionIndex!: number;
  
  @Field()
  @prop({ required: true })
  progressionName!: string;

  @Field(() => String)
  @prop({ required: true, default: new Date() })
  progressionDate!: Date;

  @Field()
  @prop()
  notes?: string;
}
```
```typescript
@plugin(AutoIncrementSimple, [{ field: 'progressionIndex' }])
@InputType()
export class ProgressionInput implements Partial<Progression> {
  @Field(() => ObjectIdScalar)
  positionId: mongoose.Schema.Types.ObjectId;

  @Field({ nullable: false })
  progressionIndex!: number;

  @Field({ nullable: false })
  progressionName!: string;

  @Field(() => String)
  progressionDate!: Date;

  @Field({ nullable: true })
  notes?: string;
}
```
error message in graphql playground: **\"ProgressionInput.progressionIndex\" of required type \"Float!\" was not provided.",**

**Comment by hasezoey** [comment]
if `Progression` is your typegoose model, then you have to apply the plugin there, otherwise, add `@prop` decorators to `ProgressionInput`

**Reply by Sed9yDataBank** [reply]
`progression` is a subdocument, the `progressionIndex` is **required** in the subdocument class. 
The `progressionInput` shows an error when I don't add the progressionIndex property which should be auto incremented not added manually by user. In graphql playground ,the mutation it shows an error when I don't give a value to progressionIndex.

**Reply by hasezoey** [reply]
the properties for any function of this plugin need to be existent in the mongoose schema, it will not add them, meaning this plugin will error if the property does not exist on the schema
and the plugin `AutoIncrementSimple` is currently applied to `ProgressionInput`, which does not contain any `@prop`, which means there are no properties within the resulting schema
